Sprint2 12 create game service

lrapp1 opened the pull request:
resolves #12 #30 #29 #28 #25 (lol)

Putting up what I have. I think the endpoints are right but since getGames() currently returns nothing our game-list does not appear on-screen. Was hoping the homies had some advice on how to proceed.

![Screen Shot 2022-10-20 at 5 03 46 PM](https://user-images.githubusercontent.com/95186966/197057731-4fccf6d9-9ae3-4757-a867-0dbfa08728c2.png)

oldskoolfan commented on app/Http/Controllers/GameController.php:
```diff
@@ -4,13 +4,17 @@
+            'greeting'=>'Hello World'
         ]);
     }
 
```
the data returned in getGames can be returned in the GameService::getGames() method for now.

also getGameById should just call your service method to get the game by id, then return the view. Basically it can look almost identical to the gameList function.

mdigrandi replied:
I don't think there should there be a "?" in front of int? In the getGamesById function it is just "int $id". Not sure though!

oldskoolfan commented on app/Services/GameService.php:
```diff
@@ -0,0 +1,47 @@
+        ];
+    }
+
+    function searchGamesByTitle (string $searchTitle)
```
the idea for this function is to return a game, basically it's getGameById but by title.

oldskoolfan reviewed (commented):
this is really close, great job! just a few minor tweaks needed.

lrapp1 commented:
Okay sooooooooo @JessEpley-dev @12bitsample @mdigrandi @oldskoolfan I am having a very hard time with this, I rewatched the class video from last night and read Andrew's comments but I still feel like I am doing something wrong. I think this needs a new pair of eyes.

12bitsample reviewed (commented):
Hey Laura, I can't seem to find anything specifically wrong with what you've changed but since you mentioned you are having trouble with the routes, you might want to consider double checking the routes and making sure nothing was changed that they depend on? Just a thought, hopefully that helps!

oldskoolfan commented on app/Http/Controllers/GameController.php:
```diff
@@ -4,37 +4,16 @@
-        ];
-    }
-
-    public function getGameById(int $id)
```
we still need this endpoint because we have a game-info route that calls this

oldskoolfan commented on app/Services/GameService.php:
```diff
@@ -0,0 +1,60 @@
+namespace App\Services;
+
+class GameService {
+    public function __construct( private GameService $gameService)
```
the GameService shouldn't have a dependency on itself

oldskoolfan commented on app/Services/GameService.php:
```diff
@@ -0,0 +1,60 @@
+            if ($game->id === $id) {
+                return $game;
+            }
+            return view('games', ['games'=>$this->gameService->getGamesById($id)]);
```
this method should return a single game, not a view. the controller calling this method will handle be responsible for the view.

oldskoolfan commented on app/Services/GameService.php:
```diff
@@ -0,0 +1,60 @@
+        $orderBy;
+        $direction;
+        $limit;
+        return view('game-list', [
```
this method should return a list of games, not a view. the controller calling this method will handle be responsible for the view.

oldskoolfan commented on app/Services/GameService.php:
```diff
@@ -0,0 +1,51 @@
+             }
+             return view('game-info', ['game'=>$this->gameService->searchGamesByTitle()]);
+            }
+        return view('games', ['games'->$this->gameService->getGames()]);
```
this function should not be returning a view, that is the controller's job

lrapp1 commented:
- Code coverage is still failing but I think that can be resolved in Michelle's test?
- GameService has all required methods
- Game Controller depends on gameService to route to game-list (this is a current bug results in a 404 although I think the in-line logic is correct)

Updated game-list
<img width="1508" alt="Screen Shot 2022-10-27 at 2 17 04 PM" src="https://user-images.githubusercontent.com/95186966/198370781-76304a18-b3e6-4177-96d8-ac56ff3fd802.png">

Updated game-info
<img width="1508" alt="Screen Shot 2022-10-27 at 2 29 23 PM" src="https://user-images.githubusercontent.com/95186966/198370953-4b422864-dfb2-430a-9e49-0641c68b09b3.png">

Wish-List
<img width="1508" alt="Screen Shot 2022-10-27 at 2 30 07 PM" src="https://user-images.githubusercontent.com/95186966/198371008-bcd2c5d7-e16d-473e-8804-eced56110288.png">

404 bug
<img width="1508" alt="Screen Shot 2022-10-27 at 2 25 35 PM" src="https://user-images.githubusercontent.com/95186966/198371079-ee938ef5-f7d0-45f4-88b3-a9127afd1a82.png">

oldskoolfan reviewed (approved):
great job! thank you for addressing the bugs as well!

oldskoolfan reviewed (approved):
great job on this! i appreciate the effort
